### PR TITLE
Change cluster preference from NamespacedResource to Resource

### PR DESCRIPTION
**Pull request by RoniKishner**
##### Short description:
Change the cluster preference resource inheritance from NamespacedResource to Resource

##### More details:
Should have been Resource to begin with since its a global cluster Resource

### CODE REVIEW

**Comment by RoniKishner** [comment]
Verified!

**Comment by RoniKishner** [comment]
!verified

**Comment by myakove** [comment]
@RoniKishner Please fix the error (api group should be `instancetype.kubevirt.io`)
for:

```
virtualmachineclusterinstancetypes    vmclusterinstancetype,vmclusterinstancetypes,vmcf,vmcfs   instancetype.kubevirt.io/v1alpha2              false   VirtualMachineClusterInstancetype
virtualmachineclusterpreferences      vmcp,vmcps                                                instancetype.kubevirt.io/v1alpha2              false   VirtualMachineClusterPreference
virtualmachineinstancetypes           vminstancetype,vminstancetypes,vmf,vmfs                   instancetype.kubevirt.io/v1alpha2              true    VirtualMachineInstancetype
virtualmachinepreferences             vmpref,vmprefs,vmp,vmps                                   instancetype.kubevirt.io/v1alpha2              true    VirtualMachinePreference

```

File: ocp_resources/virtual_machine_cluster_preferences.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
-from ocp_resources.resource import NamespacedResource
+from ocp_resources.resource import Resource
 
 
-class VirtualMachineClusterPreference(NamespacedResource):
-    api_group = NamespacedResource.ApiGroup.INSTANCE_TYPE_KUBEVIRT_IO
+class VirtualMachineClusterPreference(Resource):
+    api_group = Resource.ApiGroup.INSTANCE_TYPE_KUBEVIRT_IO
+    api_version = Resource.ApiVersion.V1ALPHA2
```
Comment (myakove) [inline ocp_resources/virtual_machine_cluster_preferences.py:6]: api_version is not needed. we take it automaticlly.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/virtual_machine_cluster_preferences.py (hunk 1 above)
Comment (RoniKishner) [reply ocp_resources/virtual_machine_cluster_preferences.py]: Are you sure it's not needed? the test passes only when I add it...

**Comment by myakove** [comment]
@RoniKishner please rename INSTANCE_TYPE_KUBEVIRT_IO to INSTANCETYPE_KUBEVIRT_IO

**Comment by myakove** [comment]
@RoniKishner please rebase on https://github.com/RedHatQE/openshift-python-wrapper/pull/870 
it should solve CI issues.

File: ocp_resources/resource.py [hunk 2]
```diff
@@ -317,6 +317,7 @@ class ApiVersion:
         V1 = "v1"
         V1BETA1 = "v1beta1"
         V1ALPHA1 = "v1alpha1"
+        V1ALPHA2 = "v1alpha2"
```
Comment (myakove) [inline ocp_resources/resource.py:320]: `V1ALPHA2 = "v1alpha2"` not used, please remove it.

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by RoniKishner** [comment]
!verified

**Review by myakove — APPROVED** [review]

**Comment by myakove** [comment]
!cherry-pick v4.12

**Pull request merged**